### PR TITLE
mds: warn if insufficient standbys exist

### PR DESCRIPTION
**Pull request by batrick**

### CODE REVIEW

File: src/mon/MonCommands.h [hunk 1]
```diff
@@ -398,7 +398,8 @@ COMMAND("fs get name=fs_name,type=CephString", \
 COMMAND("fs set " \
 	"name=fs_name,type=CephString " \
 	"name=var,type=CephChoices,strings=max_mds|max_file_size"
-        "|allow_new_snaps|inline_data|cluster_down|allow_multimds|allow_dirfrags|balancer " \
+        "|allow_new_snaps|inline_data|cluster_down|allow_multimds|allow_dirfrags|balancer" \
+        "|standby_count_desired " \
```
Comment (batrick) [inline src/mon/MonCommands.h]: "desired" in particular kinda strange. I think I'll changed that to "wanted".

Comment (batrick) [inline src/mon/MonCommands.h]: @jcsp, updated to "standby_count_wanted".

File: src/mds/MDSMap.h [hunk 2]
```diff
@@ -290,6 +292,15 @@ class MDSMap {
   mds_rank_t get_max_mds() const { return max_mds; }
   void set_max_mds(mds_rank_t m) { max_mds = m; }
 
+  mds_rank_t get_standby_count_wanted() {
```
Comment (jcsp) [inline src/mds/MDSMap.h]: Method should be const

File: src/mds/FSMap.cc [hunk 3]
```diff
@@ -687,6 +696,8 @@ void FSMap::promote(
     // An existing rank is being assigned to a replacement
     info.state = MDSMap::STATE_REPLAY;
     mds_map.failed.erase(assigned_rank);
+    if (mds_map.standby_count_wanted == 0)
```
Comment (jcsp) [inline src/mds/FSMap.cc]: It's not obvious to me why this belongs here: it seems like we're only going to miss setting standby_count_wanted initially on a healthy system with one active and one standby -- if the standby dies we won't warn, unless the active has died and been replaced previously.

File: src/mds/MDSMap.h [hunk 4]
```diff
@@ -193,6 +193,7 @@ class MDSMap {
    */
 
   mds_rank_t max_mds; /* The maximum number of active MDSes. Also, the maximum rank. */
+  mds_rank_t standby_count_wanted;
```
Comment (jcsp) [inline src/mds/MDSMap.h]: Currently I think that if the user decides he doesn't like the warning and wants to reset this to zero, we're going to go ahead and set it back to 1 in the cases where it's auto set during promote().  That's probably not desired behaviour -- what if instead we defaulted/initialized standby_count_wanted to -1, and then only auto-set it to 1 from -1.  If it was zero, we should assume that is a human-set value and not modify it for them.

**Comment by jcsp** [comment]
Please add docs for the new health warning in doc/cephfs/health-messages.rst (and probably mention how to configure it in doc/cephfs/standby.rst)

**Comment by batrick** [comment]
@jcsp I've implemented your suggestions. I feel a little guilty about changing `standby_count_wanted` in a getter but it seemed like the right spot. Do you think it's okay?

I'm writing tests for this now.

**Comment by batrick** [comment]
Oh, this doesn't work the way I thought it would... disregard. Need to make more adjustments.

**Comment by jcsp** [comment]
@batrick this is waiting for updates from you, right?

**Comment by batrick** [comment]
Yes, I haven't forgotten... just on the back burner right now.

**Comment by batrick** [comment]
@jcsp this is ready for another review.

**Comment by batrick** [comment]
Rebase on current master due to conflicts.

**Comment by batrick** [comment]
@jcsp friendly reminder ping

**Comment by jcsp** [comment]
retest this please (jenkins)

**Pull request merged**